android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java

```diff
@@ -31,7 +39,44 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
       return;
     }
 
+    ExponentDB.experienceIdToExperience(remoteMessage.getData().get("experienceId"), new ExponentDB.ExperienceResultListener() {
+      @Override
+      public void onSuccess(ExperienceDBObject experience) {
+        try {
+          JSONObject manifest = new JSONObject(experience.manifest);
+          if (ABIVersion.toNumber(manifest.getString("sdkVersion")) >= 39) {
+            dispatchToNextNotificationModule(remoteMessage);
+            return;
+          }
+
+          JSONObject androidSection = manifest.optJSONObject("android");
+          if (androidSection != null) {
+            boolean useNextNotificationsApi = androidSection.optBoolean("useNextNotificationsApi", false);
+            if (useNextNotificationsApi) {
+              dispatchToNextNotificationModule(remoteMessage);
+              return;
+            }
+          }
```

Comment:
Could we scope this code to only run in SDK38? Someone in theory could push an update to an older-SDK-ed app with a custom field.